crates/pipeline_manager/src/db/mod.rs

```diff
@@ -838,71 +818,67 @@ impl Storage for ProjectDB {
-            &[&config_name, &config_description, &config, &project_id.map(|id| id.0), &config_id.0])
+        let row = txn.query_opt("UPDATE pipeline SET version = version + 1, name = $1, description = $2, config = COALESCE($3, config), program_id = $4 WHERE id = $5 RETURNING version",
+            &[&pipline_name, &pipeline_description, &config, &program_id.map(|id| id.0), &pipeline_id.0])
             .await
-            .map_err(|e| ProjectDB::maybe_project_id_foreign_key_constraint_err(EitherError::Tokio(e), project_id))?;
+            .map_err(|e| ProjectDB::maybe_unique_violation(EitherError::Tokio(e)))
+            .map_err(|e| ProjectDB::maybe_program_id_foreign_key_constraint_err(e, program_id))?;
         txn.commit().await?;
         match row {
             Some(row) => Ok(Version(row.get(0))),
-            None => Err(DBError::UnknownConfig(config_id).into()),
+            None => Err(DBError::UnknownPipeline(pipeline_id).into()),
         }
     }
 
-    async fn delete_config(&self, config_id: ConfigId) -> AnyResult<()> {
+    async fn delete_config(&self, pipeline_id: PipelineId) -> AnyResult<()> {
         let res = self
             .pool
             .get()
             .await?
-            .execute("DELETE FROM project_config WHERE id = $1", &[&config_id.0])
+            .execute("DELETE FROM pipeline WHERE id = $1", &[&pipeline_id.0])
             .await?;
         if res > 0 {
             Ok(())
         } else {
-            Err(anyhow!(DBError::UnknownConfig(config_id)))
+            Err(anyhow!(DBError::UnknownPipeline(pipeline_id)))
         }
     }
 
-    async fn get_attached_connector_direction(&self, uuid: &str) -> AnyResult<Direction> {
+    /// Returns true if the connector of a given name is an input connector.
+    async fn attached_connector_is_input(&self, name: &str) -> AnyResult<bool> {
         let row = self
             .pool
             .get()
             .await?
             .query_one(
-                "SELECT is_input FROM attached_connector WHERE uuid = $1",
-                &[&uuid],
+                "SELECT is_input FROM attached_connector WHERE name = $1",
+                &[&name],
             )
             .await?;
 
-        if row.get(0) {
-            Ok(Direction::Input)
-        } else {
-            Ok(Direction::Output)
-        }
+        Ok(row.get(0))
     }
 
-    async fn new_pipeline(
-        &self,
-        config_id: ConfigId,
-        config_version: Version,
-    ) -> AnyResult<PipelineId> {
-        let row = self.pool.get().await?.query_one(
-                "INSERT INTO pipeline (config_id, config_version, shutdown, created) VALUES($1, $2, false, extract(epoch from now())) RETURNING id",
-            &[&config_id.0, &config_version.0])
-            .await
-            .map_err(|e| ProjectDB::maybe_config_id_foreign_key_constraint_err(EitherError::Tokio(e), config_id))?;
+    async fn deploy_pipeline(&self, pipeline_id: PipelineId) -> AnyResult<()> {
```

Comment:
This suggests that it is better to have a status field that represents different states like Pending, Created etc. Right now, "new pipeline" has shutdown = false and created = NULL. A deployed pipeline has shutdown = false and created != NULL. A deleted pipeline has shutdown = true. Seems clearer to just have one status field.